File: plugin/src/main/java/io/xol/z/plugin/game/SpawnPoints.java
//
// This file is a part of the XolioZ Mod for Chunk Stories
// Check out README.md for more information
// Website: https://chunkstories.xyz
// Github: https://github.com/Hugobros3/xolioz
//

package io.xol.z.plugin.game;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import io.xol.z.plugin.XolioZPlugin;

// Copyright 2014 XolioWare Interactive

public class SpawnPoints {

	static List<int[]> points = new ArrayList<int[]>();

	public static void load() {
		points.clear();
		File file = getFile();
		try {
			InputStream ips = new FileInputStream(file);
			InputStreamReader ipsr = new InputStreamReader(ips, "UTF-8");
			BufferedReader br = new BufferedReader(ipsr);
			String ligne;
			while ((ligne = br.readLine()) != null) {
				String[] data = ligne.split(" ");
				if (data.length >= 3) {
					int[] coords = { Integer.parseInt(data[0]), Integer.parseInt(data[1]), Integer.parseInt(data[2]) };
					points.add(coords);
				}
			}
			br.close();
		} catch (IOException e) {
			e.printStackTrace();
		}
		System.out.println("[DogeZ]" + points.size() + " spawn points loaded.");
	}

	private static File getFile() {
		File file = new File(XolioZPlugin.pluginFolder + "spawnPoints.dz");
		if (!file.exists())
			try {
				file.createNewFile();
			} catch (IOException e1) {
			}
		return file;
	}

	public static int[] getRandomSpawn() {
		Random random = new Random();
		return points.get(random.nextInt(points.size()));
	}
}

File: plugin/src/main/java/io/xol/z/plugin/game/SpawnPoints.kt
//
// This file is a part of the XolioZ Mod for Chunk Stories
// Check out README.md for more information
// Website: https://chunkstories.xyz
// Github: https://github.com/Hugobros3/xolioz
//

package io.xol.z.plugin.game

import com.google.gson.GsonBuilder
import com.google.gson.reflect.TypeToken
import io.xol.z.plugin.XolioZPlugin
import java.io.File
import java.io.FileNotFoundException
import java.util.*
import java.util.ArrayList

class SpawnPoints(val plugin: XolioZPlugin) {

    private var points: MutableList<SpawnPoint> = ArrayList()

    val file: File = File(plugin.folder().absolutePath + "/spawnPoints.json")

    init {
        file.parentFile.mkdirs()
        load()
    }

    fun randomSpawn(): SpawnPoint {
        val random = Random()
        return points[random.nextInt(points.size)]
    }

    fun load() {
        try {
            if(!file.exists()) throw FileNotFoundException(file.absolutePath)

            val data = file.readText()
            points = gson.fromJson(data, object : TypeToken<ArrayList<SpawnPoint>>() { }.type)

        } catch (e: Exception) {
            plugin.logger.warning("Spawn points file not found or corrupted (${e.message}), using failsafe (0,100,0)")
            points = ArrayList()
            points.add(SpawnPoint(0, 100, 0))
        }

        plugin.logger.info("${points.size} spawn points loaded.")
    }

    fun save() {

    }

    companion object {
        val gson = GsonBuilder().setPrettyPrinting().create()
    }
}
